### PR TITLE
Refactor/oop

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/bookstore
Because OO code is good code

### What Changed?
Removed extraneous class dependencies
Moved behavior into proper classes
Renamed confusing/conflicting methods

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -13,22 +13,23 @@ class Order < ActiveRecord::Base
 
   delegate :name, to: :order_status, prefix: true
 
-  scope :pending, -> { where(order_status_id: 1) }
+  scope :pending, -> { where(order_status_id: OrderStatus.find_by(name: "Pending")) }
```
Comment (rob-m-andrews) [inline app/models/order.rb]: Is `OrderStatus` just a lookup table? It seems like you could cache all of these statuses once at some level in the application and store them in a hash for easy lookup. I assume that ActiveRecord is pretty good at caching, so maybe this is fine.

Comment (TylerRockwell) [inline app/models/order.rb]: I'm going to do something along those lines. I have bigger priorities on this project at the moment, but I plan to rewrite the way OrderStatuses are handled throughout the app. Annie and I discussed a better solution yesterday.

File: app/models/order_item.rb [hunk 2]
```diff
@@ -1,18 +1,13 @@
 class OrderItem < ActiveRecord::Base
-  belongs_to :book
```
Comment (rob-m-andrews) [inline app/models/order_item.rb]: I noticed you removed the `belongs_to` and added a `book_title` property. The generated schema still shows order_item having a `book_id`, is that intentional?

File: app/models/order.rb (hunk 1 above)
Comment (rob-m-andrews) [inline app/models/order.rb]: Sounds good to me!

**Comment by rob-m-andrews** [comment]
@TylerRockwell This is looking good! I had one question about the `belongs_to` relationship

File: app/models/order_item.rb (hunk 2 above)
Comment (TylerRockwell) [inline app/models/order_item.rb]: No, I must have forgotten to remove it. I'll add a migration now.

**Comment by TylerRockwell** [comment]
@rob-m-andrews I removed that field as it was no longer being used. Thanks for looking this over!

File: app/models/order_item.rb (hunk 2 above)
Comment (rob-m-andrews) [inline app/models/order_item.rb]: Yeeeh! Nice :100:

**Comment by rob-m-andrews** [comment]
Looks good to me, @TylerRockwell :shipit:

**Comment by TylerRockwell** [comment]
Awesome. Thanks again, @rob-m-andrews !

**Pull request merged**